src/Cmd/Add.cc

```diff
@@ -124,8 +127,15 @@ addDependencyToManifest(
     }
   }
 
-  auto data = toml::parse(getManifestPath());
-  auto& deps = toml::find<toml::table>(data, "dependencies");
+  // parse with ordered_type_config to keep the order of the tables
+  auto data = toml::parse<toml::ordered_type_config>(getManifestPath());
+
+  // Ensure the "dependencies" key exists and is a table for correct
+  // serialization
+  if (!data["dependencies"].is_table()) {
+    data["dependencies"] = toml::table{};
+  }
```

Comment:
If this isn't a table, shouldn't we throw an exception? Overwriting the existing, incorrect data without notifying the user is kind of aggressive, I think.